pkg/apis/kudo/v1beta1/parameter_types_helpers.go

```diff
@@ -36,6 +36,11 @@ func (p *Parameter) ValidateDefault() error {
 		return fmt.Errorf("parameter %q has an invalid default value: %v", p.Name, err)
 	}
 	if p.IsEnum() {
+		if *p.Default == "" && !p.IsRequired() {
```

Comment:
uff I am lost here, so what does it mean for default to be nil or empty string? I would expect that HasDefault already covers this case

Reply:
After second thoughts, I'll retract this PR. I wanted to fix something that I noticed with the Cassandra Params:

```
  - name: DISK_OPTIMIZATION_STRATEGY
    default: ""
    enum:
      - "ssd"
      - "spinning"
```

These parameters are not set by default and had a "default" of an empty string, although the enum only allows "ssd" and "spinning". The validation now complained about the "" for default, because it's not in the list of valid values.

The actual solution though, is to remove the default and set the parameter to not-required:
```
  - name: DISK_OPTIMIZATION_STRATEGY
    required: false
    enum:
      - "ssd"
      - "spinning"
```

This is valid and solves the problem. We still have the problem that it's very hard (if not impossible) to unset a parameter with the CLI, but that's a different issue